src/ucm/mmap/install.c

```diff
@@ -104,23 +110,43 @@ void ucm_fire_mmap_events(int events)
 
 ucs_status_t ucm_mmap_test_events(int events, int *out_events)
 {
+    ucm_mmap_test_event_data_t data = {.out_events = out_events};
+    int native_events;
+    int event;
     ucm_event_handler_t handler;
 
+    native_events = (events & UCM_EVENT_VM_MAPPED   ? UCM_NATIVE_EVENT_VM_MAPPED   : 0) |
+                    (events & UCM_EVENT_VM_UNMAPPED ? UCM_NATIVE_EVENT_VM_UNMAPPED : 0) |
+                    (events & ~(UCM_EVENT_VM_MAPPED | UCM_EVENT_VM_UNMAPPED));
```

Comment:
don't understand this line, we get here only VM_MAPPED and VM_UNMAPPED were not set, so what's the purpose to remove those bits?

Reply:
we are checking for native events only: VM_**** are expanded into set of native events which should be fired, and moreover: **all** native events must be fired for specific aggregate event. fire_event call fires multiple events for every aggregate event which makes impossible to identify missing hooks